src/System.IO.FileSystem.DriveInfo/src/System/IO/DriveInfo.Unix.cs

```diff
@@ -82,8 +128,18 @@ public long TotalSize
             [SecuritySafeCritical]
             get
             {
-                Interop.libc.statfs data = Interop.libc.GetStatFsForDriveName(Name);
-                return (long)data.f_bsize * (long)data.f_blocks;
+                Interop.Sys.MountPointInformation mpi;
+                int result = Interop.Sys.GetSpaceInfoForMountPoint(Name, out mpi);
+                if (result == 0)
+                    return mpi.TotalSize;
+                else
+                {
+                    Interop.ErrorInfo err = Interop.Sys.ConvertErrorPlatformToPal(result).Info();
```

Comment:
I added a helper to check the result on non-zero; that should de-dupe the code and make it cleaner